### PR TITLE
[I18N] add tabs for navigating between forms

### PR DESCRIPTION
**Pull request by mfrachet**
### What does it do?

- Iterating on the i18n ModalEdit, switching the input and positioning elements
- Suggests a declarative tabs navigation API using indexes. Selecting the first Tab will display the content of the first TabPanel.

If needed, we can add keyboard navigation (using left / right arrow) quite easily to this one.

```jsx
<TabsNav defaultSelection={0} label="This is an aria label" id="i18n-settings-tabs">
  <Tabs position="right">
    <Tab>Base settings</Tab>
    <Tab>Advanced settings</Tab>
  </Tabs>

  <TabsPanel>
    <TabPanel>
      <BaseForm />
    </TabPanel>
    <TabPanel>
      <AdvancedForm />
    </TabPanel>
  </TabsPanel>
</TabsNav>
```

### CODE REVIEW

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js [hunk 1]
```diff
@@ -31,51 +39,60 @@ const ModalEdit = ({ localeToEdit, onClose, locales }) => {
...
+                      <Tab>{formatMessage({ id: getTrad('Settings.locales.modal.advanced') })}</Tab>
+                    </Tabs>
+
+                    <TabsPanel>
+                      <TabPanel>
+                        <BaseForm options={options} defaultOption={defaultOption} />
+                      </TabPanel>
+                      <TabPanel>advanced</TabPanel>
+                    </TabsPanel>
+                  </TabsNav>
```
Comment (mfrachet) [inline packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js:91]: ℹ️ This is a suggestion to create navigable tabs. We can also add the power to navigate with the keyboard between them quite easily if necessary 😄

File: packages/strapi-helper-plugin/lib/src/components/Tabs/TabComponents.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  margin-left: ${props => (props.position === 'right' ? 'auto' : 0)};
+  border-bottom: 1px solid ${props => props.theme.main.colors.brightGrey};
+`;
+
+export const TabButton = styled.button`
+  height: 3.8rem;
+  font-size: 1.2rem;
+  letter-spacing: 0.7px;
+  text-transform: uppercase;
+  border-bottom: 2px solid ${props => (props['aria-selected'] ? '#007eff' : 'transparent')};
```
Comment (HichamELBSI) [inline packages/strapi-helper-plugin/lib/src/components/Tabs/TabComponents.js:24]: you can use the mediumBlue color of the theme here 😉 

```js
   border-bottom: 2px solid ${props => (props['aria-selected'] 
      ? props.theme.main.colors.mediumBlue 
      : 'transparent')};
```

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/BaseForm.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+const BaseForm = ({ options, defaultOption }) => {
+  const { formatMessage } = useIntl();
+  const { values, handleChange } = useFormikContext();
+
+  return (
+    <Row>
+      <Col>
+        <span id="locale-code">
+          <Label htmlFor="">
```
Comment (HichamELBSI) [inline packages/strapi-plugin-i18n/admin/src/components/ModalEdit/BaseForm.js:19]: is this `htmlFor` empty value wanted?

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js [hunk 4]
```diff
@@ -31,51 +43,60 @@ const ModalEdit = ({ localeToEdit, onClose, locales }) => {
...
+                  style={{
+                    fontSize: '1.8rem',
+                    height: '65px',
+                    fontWeight: 'bold',
+                    alignItems: 'center',
+                    marginBottom: '-39px',
+                    paddingTop: '16px',
+                  }}
+                >
+                  Configurations
```
Comment (HichamELBSI) [inline packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js:73]: I think this should be a translated message.

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js [hunk 5]
```diff
@@ -31,51 +43,60 @@ const ModalEdit = ({ localeToEdit, onClose, locales }) => {
...
+                    <Tabs position="right">
+                      <Tab>{formatMessage({ id: getTrad('Settings.locales.modal.base') })}</Tab>
+                      <Tab>{formatMessage({ id: getTrad('Settings.locales.modal.advanced') })}</Tab>
+                    </Tabs>
+
+                    <TabsPanel>
+                      <TabPanel>
+                        <BaseForm options={options} defaultOption={defaultOption} />
+                      </TabPanel>
+                      <TabPanel>advanced</TabPanel>
```
Comment (HichamELBSI) [inline packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js:93]: Same here.

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/BaseForm.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+BaseForm.propTypes = {
+  options: PropTypes.arrayOf(
+    PropTypes.shape({ value: PropTypes.number.isRequired, label: PropTypes.string.isRequired })
+  ).isRequired,
+  defaultOption: PropTypes.shape({
+    value: PropTypes.number.isRequired,
+    label: PropTypes.string.isRequired,
+  }),
```
Comment (HichamELBSI) [inline packages/strapi-plugin-i18n/admin/src/components/ModalEdit/BaseForm.js:70]: Can you use `exact` propTypes for both props

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js (hunk 1 above)
Comment (HichamELBSI) [reply packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js:95]: I like that API 👍🏽 
What are your thoughts on the "Compound Components"
https://kentcdodds.com/blog/compound-components-with-react-hooks

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

**Review by HichamELBSI — CHANGES_REQUESTED** [review]
Good job 😄

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/BaseForm.js (hunk 3 above)
Comment (mfrachet) [reply packages/strapi-plugin-i18n/admin/src/components/ModalEdit/BaseForm.js:19]: The `Label` component requires a `htmlFor` (as a mandatory props). In a normal situation, this htmlFor would point to the id of the select input provided by react-select. 

However, it looks that react-select doesn't allow to pass down an id, but allows to pass an `aria-labelledby` props. This props references an "id" that describes it's content. (the id is the one that you see line 18).

So:

- I had to put the htmlFor because of propTypes
- And I'm referencing the label in the "inverse" way by using aria-labelledby 😊

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js (hunk 5 above)
Comment (mfrachet) [reply packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js:93]: Concerning this one, it's the place where the `<AdvancedForm />` will go later. It's not supposed to be translated, it's the place where the form will take place 😊

**Comment by mfrachet** [comment]
Thank you @HichamELBSI for the review and comments 🙏🏻 

Concerning the question:

> What are your thoughts on the "Compound Components pattern"

The Tabs component is exactly one of them 😄 . When building things, I like to be explicit and to provide clarity. What I like about compound components, even if they hide complexity somehow, is that I can directly understand what is their intent, and what they aim to do. I don't have to scroll up and down in a file, nor jump between files.

They also help to maintain composition. For instance, we could have written this component using something like:

```jsx
const tabs = [{label: 'Base settings', Component: BaseComponent}];

return <Tabs tabs={tabs} />
```

But with this approach, it's not possible: what if I want to display additional information, only for the "Base" tab? We can eventually add a new prop in this scenario. 

```jsx
const tabs = [{label: 'Base settings', Component: BaseComponent, additionalData: 'some data for base settings'}];

return <Tabs tabs={tabs} />
```

But this is one more prop that we have to manage in this component and that might have an impact on later usage: the more props, the more logic, the harder it is to maintain 😄 .

There are tons of reasons why I love them tbh 😊 . If you're interested, I've written down some of my ideas about that some time ago:

- https://dev.to/mfrachet/a-note-on-composing-components-with-react-5ee4
- https://mfrachet.github.io/the-n-props-syndrome

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js (hunk 5 above)
Comment (HichamELBSI) [reply packages/strapi-plugin-i18n/admin/src/components/ModalEdit/index.js:95]: My bad 🙈

File: packages/strapi-helper-plugin/lib/src/components/Tabs/TabComponents.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import styled from 'styled-components';
+
+export const TabNavRaw = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+`;
+
+export const TabsRaw = styled.div`
```
[the remark this answers is not in the document]
Comment (HichamELBSI) [reply packages/strapi-helper-plugin/lib/src/components/Tabs/TabComponents.js:9]: Maybe you forgot this comment?

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

File: packages/strapi-helper-plugin/lib/src/components/Tabs/TabComponents.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (mfrachet) [reply packages/strapi-helper-plugin/lib/src/components/Tabs/TabComponents.js]: Wooops! Thank you 😅

File: packages/strapi-plugin-i18n/admin/src/components/ModalEdit/BaseForm.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+      <Col>
+        <span id="locale-code">
+          <Label htmlFor="">
+            {formatMessage({
+              id: getTrad('Settings.locales.modal.edit.locales.label'),
+            })}
+          </Label>
+        </span>
+
+        <Select
```
Comment (soupette) [inline packages/strapi-plugin-i18n/admin/src/components/ModalEdit/BaseForm.js:26]: You will need to apply some design to this component but you can do it in another PR.

**Review by soupette — APPROVED** [review]
LGTM
